bookkeeper-server/src/main/java/org/apache/bookkeeper/client/BookKeeper.java

```diff
@@ -141,9 +152,25 @@ public Builder setStatsLogger(StatsLogger statsLogger) {
             return this;
         }
 
+
+        public Builder dnsResolver(DNSToSwitchMapping dnsResolver) {
+            this.dnsResolver = dnsResolver;
+            return this;
+        }
+
+        public Builder requestTimer(HashedWheelTimer requestTimer) {
```

Comment:
@jvrao I have created https://issues.apache.org/jira/browse/BOOKKEEPER-942 in order to work on public API and remove this direct dependency